Assignment 3.Rmd

```diff
@@ -128,28 +133,103 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
+Z2 <- tidyr::gather(Z1, "week", "motivation", 1:5)
+Z3 <- Z2 %>% group_by(week, cluster)
+Z4 <- summarise(Z3, avg=mean(motivation))
+Z4$week <- as.numeric(Z4$week)
+Z4$cluster <- as.factor(Z4$cluster)
+ggplot(Z4, aes(week, avg, col = cluster)) + geom_line() + xlab("Week") + ylab("Average Motivation")
+
+```
+
+```{r}
+Z5 <- dplyr::count(Z1, cluster)
+```
+I think the latter cluster grouping is more informative because it is more specific.
+
 ##Part II
 
 Using the data collected for Assignment 2 (which classes students were in), cluster the students, then redraw the graph of the class but color the students according the cluster they are in.
 
+```{R}
+D1 <- read.csv("hudk4050-classes.csv", header = TRUE)
+D1$Class.1 <- gsub(pattern = " ",replacement = "", x = D1$Class.1)
+D1$Class.2 <- gsub(pattern = " ",replacement = "", x = D1$Class.2)
+D1$Class.3 <- gsub(pattern = " ",replacement = "", x = D1$Class.3)
+D1$Class.4 <- gsub(pattern = " ",replacement = "", x = D1$Class.4)
+D1$Class.5 <- gsub(pattern = " ",replacement = "", x = D1$Class.5)
+D1$Class.6 <- gsub(pattern = " ",replacement = "", x = D1$Class.6)
+D1 <- filter(D1, D1$First.Name!="ZIMO")
+D1 = data.frame(D1) %>% unite(Full.Name,First.Name,Last.Name,remove=TRUE,sep=" ")
+fit2 <- kmodes(D1,4)
+D2 <- data.frame(D1,fit2$cluster)
+#because all students are in the course HUDK4050
+D3 <- D1 %>% gather(Class, Course, -Full.Name) %>% filter(Course!="") %>% filter(Course!="HUDK4050") %>% dplyr::select(1,3)
+D3$Count <- 1
+D4 <- tidyr::spread(D3, Course, Count, fill=0)
+D4 = data.frame(D4)
+D4 <- D4 %>% remove_rownames %>% column_to_rownames(var="Full.Name")
+D4 <- as.matrix(D4)
+D5 <- t(D4)
+D6 <- D4 %*% D5
+g <- graph_from_adjacency_matrix(D6, mode = "directed", diag = FALSE)
+plot(g, vertex.color = D2$fit2.cluster)
+
+```
+
+
+
 ##Part III
 
 In class activity 6 you clustered students in the class by the answers to a questionaire. Create a visualization that shows the overlap between these clusters and the clusters generated in part II.
 
 ## Please render your code as an .html file using knitr and Pull Resquest both your .Rmd file and .html files to the Assignment 3 repository.
 
+```{R}
+A1 <- read.csv("HUDK405019-clustering.csv", header = TRUE, stringsAsFactors = FALSE)
+A1 = data.frame(A1) %>% unite(Full.Name,First.Name,Last.Name,remove=TRUE,sep=" ")
+row.names(A1) <- A1$Full.Name
+A1$Full.Name <- NULL
+A2 <- dplyr::select(A1, 1:11)
+A3 <- lapply(A2, function(x)gsub("[A-z]","",x))
+# convert all blank cells to NAs
+A3[A3==""] <- NA
+A4 <- na.omit(A3)
+A4 <- data.frame(A4)
+# convert all columns to numeric
+A4 <- A4 %>% mutate_all(funs(as.numeric(levels(.))[.]))
+A4 <- as.data.frame(scale(A4))
+fit3 <- kmeans(na.omit(A4),4)
+A5 <- data.frame(fit3$cluster)
+A6 <- dplyr::select(D2, 8)
+A7 <- merge(A5, A6)
+A7 <- na.omit(A7)
+my_table <- table(A7$fit3.cluster,A7$fit2.cluster)
+my_table
+mosaicplot(my_table, color = c(8,4,5), main = "overlay", ylab = "kmean", xlab = "kmode")
+
```

Comment:
Overall, great job on the coding part of this assignment but you may need to work on the analysis aspect of it. Nonetheless, keep up the good work!